Zend/zend_language_parser.y

```diff
@@ -547,12 +548,16 @@ unset_variable:
 ;
 
 function_declaration_statement:
-	function returns_ref T_STRING backup_doc_comment '(' parameter_list ')' return_type
-	backup_fn_flags '{' inner_statement_list '}' backup_fn_flags
-		{ $$ = zend_ast_create_decl(ZEND_AST_FUNC_DECL, $2 | $13, $1, $4,
-		      zend_ast_get_str($3), $6, NULL, $11, $8, NULL); CG(extra_fn_flags) = $9; }
+    function returns_ref T_STRING backup_doc_comment '(' parameter_list ')' return_type
+    backup_fn_flags function_body backup_fn_flags
+        { $$ = zend_ast_create_decl(ZEND_AST_FUNC_DECL, $2 | $11, $1, $4,
+            zend_ast_get_str($3), $6, NULL, $10, $8, NULL); CG(extra_fn_flags) = $9; }
 ;
 
+function_body:
+	'{' inner_statement_list '}' { $$ = $2; }
+	| T_DOUBLE_ARROW expr ';'  { $$ = zend_ast_create(ZEND_AST_RETURN, $2); }
```

Comment:
This may have subtle, or not so subtle, strange side effects when it comes to generators ...

You're actually compiling `return yield [from] ...;`

```
krakjoe@Fiji:/opt/src/php-src$ sapi/phpdbg/phpdbg -p* yield.php
function name: (null)
L1-6 {main}() /opt/src/php-src/yield.php - 0x7ff35ec030f0 + 1 ops
 L6    #0     RETURN<-1>              1                                                             

function name: read
L2-2 read() /opt/src/php-src/yield.php - 0x7ff35ec63460 + 5 ops
 L2    #0     RECV                    1                                         $it                 
 L2    #1     GENERATOR_CREATE                                                                      
 L2    #2     YIELD_FROM              $it                                       ~0                  
 L2    #3     FREE                    ~0                                                            
 L2    #4     GENERATOR_RETURN<-1>    null                                                          

function name: read_short
L4-4 read_short() /opt/src/php-src/yield.php - 0x7ff35ec63540 + 5 ops
 L4    #0     RECV                    1                                         $it                 
 L4    #1     GENERATOR_CREATE                                                                      
 L4    #2     YIELD_FROM              $it                                       ~0                  
 L4    #3     GENERATOR_RETURN        ~0                                                            
 L4    #4     GENERATOR_RETURN<-1>    null                                                          
[Script ended normally]
```

You'll have to ask someone who uses generators what that will mean in the real world, but it certainly doesn't seem ideal.

Reply:
I just noticed that fn does the same thing, not sure if oversight ...